sample_formatted_code.do_not_merge.c

```diff
@@ -0,0 +1,562 @@
+{
+    int a = 13;
+    return a * a;
+}
+
+// BraceWrapping:SplitEmptyFunction if BreakBeforeBraces: Custom
+void empty_function(void)
+{
+}
+
+// AllowShortFunctionsOnASingleLine
+int short_function(void)
+{
+    return 1;
+}
+
+// all params fit on continuation line
+// AlignAfterOpenBracket
+static void function_with_params_split(
+        const char *key, json_t *value, idmef_alert_t *alert)
+{
+    bla();
+}
+
+// params too long to fit on one continuation line, broken apart over multiple
+// lines
+// AlignAfterOpenBracket
+int some_function_with_parms_split(uint32_t *LongNameForParameter1,
+        double *LongNameForParameter2, const float *LongNameForParameter3,
+        const struct SomeStructWithALongName LongNameForParameter4)
+{
+    int a = 3;
+    return a * a;
+}
+
+//--- switch ---
+// https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Coding_Style#switch-statements
+void switch_style()
+{
+    // IndentCaseBlocks
+    // IndentCaseLabels
+
+    // Switch statements are indented like in the following example, so the
+    // 'case' is indented from the switch
+    switch (ntohs(p->ethh->eth_type)) {
+        case ETHERNET_TYPE_IP:
+            DecodeIPV4(tv, dtv, p, pkt + ETHERNET_HEADER_LEN,
+                    len - ETHERNET_HEADER_LEN, pq);
+            break;
+    }
+
+    // Fall through cases will be commented with /* fall through */. E.g.:
+    switch (suri->run_mode) {
+        case RUNMODE_PCAP_DEV:
+        case RUNMODE_AFP_DEV:
+        case RUNMODE_PFRING:
+            /* find payload for interface and use it */
+            default_packet_size = GetIfaceMaxPacketSize(suri->pcap_dev);
+            if (default_packet_size)
+                break;
+            /* fall through */
+        default:
+            default_packet_size = DEFAULT_PACKET_SIZE;
+    }
+
+    // BraceWrapping:AfterCaseLabel if BreakBeforeBraces: Custom
+    // AllowShortCaseLabelsOnASingleLine
+    switch (a) {
+        case 13: {
+            int a = bla();
+            break;
+        }
+        case 15:
+            blu();
+            break;
+        default:
+            gugus();
+    }
+}
+
+//--- goto ---
+// https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Coding_Style#goto
+void goto_style()
+{
+    DetectFileextData *fileext = NULL;
+
+    fileext = SCMalloc(sizeof(DetectFileextData));
+    if (unlikely(fileext == NULL))
+        goto error;
+
+    memset(fileext, 0x00, sizeof(DetectFileextData));
+
+    if (DetectContentDataParse("fileext", str, &fileext->ext, &fileext->len,
+                &fileext->flags) == -1) {
+        goto error;
+    }
+
+    return fileext;
+
+error:
+    if (fileext != NULL)
+        DetectFileextFree(fileext);
+    return NULL;
+}
+
+int goto_style_nested()
+{
+    // IndentGotoLabels
+    if (foo()) {
+    label1:
+        bar();
+    }
+
+label2:
+    return 1;
+}
+
+//--- ternary style ---
+// BreakBeforeTernaryOperators - whether ? and : are on next line
```

Comment:
This one is missing in code-style. Should it be added?

Reply:
Lets leave this out. I think it's fine with simple cases.